### DISCUSSION TITLE
Deprecated fields are not visible in Graphiql

### DISCUSSION THREAD

**Discussion by pjungels-paraport** [Netflix/dgs-framework#1526] [Q&A]
Greetings!

I'm relatively new to GraphQL and the DGS framework, so I might be doing something wrong here. Anyway, I am trying to deprecate some fields in our schema using the @deprecated(reason: "reason") directive. When I add the directive, the deprecated fields are no longer visible in GraphiQL and if I try to use the deprecated field in a query I get this error message: "Field <field name> is not defined by type <type name>. Did you mean <other field name 1> or <other field name 2>?"

Do I need to do something special to see/use deprecated fields?

FYI - here are the relevant items from my build.gradle.kts:
plugins {
    ...
    id("com.netflix.dgs.codegen") version "5.7.1"
    id("org.springframework.boot") version "3.0.6"
}

dependencies {
    ...
    implementation(platform("com.netflix.graphql.dgs:graphql-dgs-platform-dependencies:6.0.3"))
    implementation("com.netflix.graphql.dgs:graphql-dgs-extended-scalars")
    implementation("com.netflix.graphql.dgs:graphql-dgs-spring-boot-starter")
    implementation("com.netflix.graphql.dgs:graphql-dgs-pagination")
    ...
}

Thanks!
Paul

**Comment by srinivasankavitha** [comment]
We don't own this in the framework. We simply use this : https://www.npmjs.com/package/graphiql
Generally deprecated fields are not meant to be discoverable, but they should show up with a warning that the field is deprecated. I see that behavior in the example I just tested.

**Reply by pjungels-paraport** [reply]
@srinivasankavitha , regarding this statement: "Generally deprecated fields are not meant to be discoverable, but they should show up with a warning that the field is deprecated." That's exactly what I was expecting. However, the fields aren't even showing up for me. Here's a quick example:

Schema snippet:
```
type Query {
    securities(
        masterSecurityIds: [Int!]!,
        enhanced: Boolean = false @deprecated(reason: "some reason")
    ): [Security]
}
```
Screenshot from GraphiQL showing the "enhanced" argument isn't available anywhere:
![image](https://github.com/Netflix/dgs-framework/assets/90796309/24a026a9-7a01-4a65-b080-2ecfa201bb05)

 Do I need to do something specific so they are visible in the UI (with the expected warning)?

**Reply by srinivasankavitha** [reply]
Not that I am aware of. When I tried it , I was able to view it just fine and I don't have any special configuration.
